pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -523,32 +526,47 @@ func findSyncStatus(status hivev1.SyncStatus, statusList []hivev1.SyncStatus) *h
 }
 
 // needToReapply determines if the provided status indicates that the resource, secret or patch needs to be re-applied
-func (r *ReconcileSyncSetInstance) needToReapply(applyTerm string, newStatus, existingStatus hivev1.SyncStatus, ssiLog log.FieldLogger) bool {
+func (r *ReconcileSyncSetInstance) needToReapply(newStatus, existingStatus hivev1.SyncStatus, ssiLog log.FieldLogger) bool {
 	// Re-apply if hash has changed
 	if existingStatus.Hash != newStatus.Hash {
-		ssiLog.Debugf("%s %s/%s (%s) has changed, will re-apply", applyTerm, newStatus.Namespace, newStatus.Name, newStatus.Kind)
+		ssiLog.Debug("hash has changed, will re-apply")
 		return true
 	}
 	// Re-apply if failure occurred
 	if failureCondition := controllerutils.FindSyncCondition(existingStatus.Conditions, hivev1.ApplyFailureSyncCondition); failureCondition != nil {
 		if failureCondition.Status == corev1.ConditionTrue {
-			ssiLog.Debugf("%s %s/%s (%s) failed last time, will re-apply", applyTerm, newStatus.Namespace, newStatus.Name, newStatus.Kind)
+			ssiLog.Debug("failed last time, will re-apply")
 			return true
 		}
 	}
-	// Re-apply if past reapplyInterval
-	if applySuccessCondition := controllerutils.FindSyncCondition(existingStatus.Conditions, hivev1.ApplySuccessSyncCondition); applySuccessCondition != nil {
-		since := time.Since(applySuccessCondition.LastProbeTime.Time)
-		if since > r.reapplyInterval {
-			ssiLog.Debugf("It has been %v since %s %s/%s (%s) was last applied, will re-apply", applyTerm, since, newStatus.Namespace, newStatus.Name, newStatus.Kind)
-			return true
-		}
+	if r.reapplyTime(existingStatus).Before(time.Now()) {
+		ssiLog.Debug("re-apply duration has elapsed, will re-apply")
+		return true
 	}
 	return false
 }
 
+func (r *ReconcileSyncSetInstance) reapplyTime(status hivev1.SyncStatus) time.Time {
+	reapplyTime := status.ReapplyTime.Time
+	if !reapplyTime.IsZero() {
+		return reapplyTime
+	}
+	applySuccessCondition := controllerutils.FindSyncCondition(status.Conditions, hivev1.ApplySuccessSyncCondition)
+	if applySuccessCondition == nil {
+		return time.Now()
+	}
+	return applySuccessCondition.LastProbeTime.Add(r.reapplyInterval)
+}
+
+func (r *ReconcileSyncSetInstance) setReapplyTime(status *hivev1.SyncStatus) {
+	wait := time.Duration((1 + 0.1 * rand.Float64()) * r.reapplyInterval.Seconds()) * time.Second
```

Comment:
Is there a reason this jitter would need to be done per item in the syncset? (and stored in the status?) Wouldn't it be simpler to apply a jitter value to the requeue time? (can still be a multiple of reapplyInterval)

Reply:
sgtm!

Reply:
The code has been simplified to only jitter the requeue time.